src/MongoDB.EntityFrameworkCore/Extensions/MongoPropertyBuilderExtensions.cs

```diff
@@ -53,65 +54,127 @@ public static class MongoPropertyBuilderExtensions
-    /// <param name="dateTimeKind">The <see cref="DateTimeKind"/> to use for the property.</param>
-    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
-    public static PropertyBuilder HasDateTimeKind(
-        this PropertyBuilder propertyBuilder,
-        DateTimeKind dateTimeKind)
+    /// <param name="name">The element name for the property.</param>
+    /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+    /// <returns>The same builder instance if the configuration was applied, <see langword="null" /> otherwise.</returns>
+    public static IConventionPropertyBuilder? HasElementName(
+        this IConventionPropertyBuilder propertyBuilder,
+        string? name,
+        bool fromDataAnnotation = false)
     {
-        propertyBuilder.Metadata.SetDateTimeKind(dateTimeKind);
+        if (!CanSetElementName(propertyBuilder, name, fromDataAnnotation))
+        {
+            return null;
+        }
+
+        propertyBuilder.Metadata.SetElementName(name, fromDataAnnotation);
         return propertyBuilder;
     }
 
     /// <summary>
-    /// Configures the <see cref="DateTimeKind"/> for the property.
+    /// Returns a value indicating whether the given document element name can be set.
     /// </summary>
     /// <param name="propertyBuilder">The builder for the property being configured.</param>
-    /// <param name="dateTimeKind">The <see cref="DateTimeKind"/> to use for the property.</param>
-    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
-    public static IConventionPropertyBuilder HasDateTimeKind(
+    /// <param name="name">The element name for the property.</param>
+    /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+    /// <returns><see langword="true" /> if the field name can be set, <see langword="false"/> if not.</returns>
+    public static bool CanSetElementName(
         this IConventionPropertyBuilder propertyBuilder,
-        DateTimeKind dateTimeKind)
+        string? name,
+        bool fromDataAnnotation = false)
+        => propertyBuilder.CanSetAnnotation(MongoAnnotationNames.ElementName, name, fromDataAnnotation);
+
+    /// <summary>
+    /// Configures the <see cref="BsonType"/> that the property is stored as when targeting MongoDB.
+    /// </summary>
+    /// <param name="propertyBuilder">The builder for the property being configured.</param>
+    /// <param name="bsonType">The <see cref="BsonType"/> to store this property as.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public static PropertyBuilder HasBsonType(
+        this PropertyBuilder propertyBuilder,
+        BsonType bsonType)
     {
-        propertyBuilder.Metadata.SetDateTimeKind(dateTimeKind);
+        ArgumentNullException.ThrowIfNull(bsonType);
```

Comment:
`BsonType` is an enum and cannot be null.

Reply:
Do we want to make this `BsonType?` so that `null` is valid and can be used to unset a previous `HasBsonType` configuration?

Reply:
Yeah, let's do that.

Reply:
Now done.